### DISCUSSION TITLE
Disable max_body_size

### DISCUSSION THREAD

**Discussion by ruspaul013** [nginx/unit#1210] [Support] [answered]
Hello!

Is there a way to disable `max_body_size` so that I can upload bigger files without getting 413 and if nginx unit can be configured to pass to php request without reading body. I know that I can set `max_body_size` to a bigger value, but I was just wondering if disable is supported.  Something similar to [nginx client_max_body_size](https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size).

Thank you.

**Comment by ac000** [comment, marked as answer]
It's not currently. But seems trivial to do so..

```diff
diff --git a/src/nxt_http_request.c b/src/nxt_http_request.c
index 425a4607..abf00c86 100644
--- a/src/nxt_http_request.c
+++ b/src/nxt_http_request.c
@@ -230,7 +230,7 @@ nxt_http_request_content_length(void *ctx, nxt_http_field_t *field,
 
             max_body_size = r->conf->socket_conf->max_body_size;
 
-            if (nxt_slow_path(n > max_body_size)) {
+            if (nxt_slow_path(max_body_size != -1 && n > max_body_size)) {
                 return NXT_HTTP_PAYLOAD_TOO_LARGE;
             }
 
````

That will let you set

```json
    "settings": {
        "http": {
            "max_body_size": -1
        }
    },
```

Quick test shows it to work...

**Reply by ruspaul013** [reply]
Thank you for the response.

**Comment by ruspaul013** [comment]
I have one more question, is there a way to pass the body direct to the php app, to avoid nginx-unit to read the body of the request?

**Reply by ac000** [reply]
I don't believe so, certainly not how things currently are.
